apps/actors/lib/actors/actor/entity.ex

```diff
@@ -214,51 +221,16 @@ defmodule Actors.Actor.Entity do
         } = state
       )
       when is_nil(current_state) do
-    payload =
-      ActorInvocation.new(
-        actor_name: name,
-        actor_system: actor_system,
-        command_name: command,
-        value: payload,
-        current_context: Context.new()
-      )
-      |> ActorInvocation.encode()
-
-    case Actors.Node.Client.invoke_host_actor(payload) do
-      {:ok, %Tesla.Env{body: ""}} ->
-        Logger.error("User Function Actor response Invocation body is empty")
-        {:error, :no_content}
-
-      {:ok, %Tesla.Env{body: nil}} ->
-        Logger.error("User Function Actor response Invocation body is nil")
-        {:error, :no_content}
-
-      {:ok, %Tesla.Env{body: body}} ->
-        with %ActorInvocationResponse{
-               updated_context: %Context{} = user_ctx
-             } = resp <- ActorInvocationResponse.decode(body) do
-          {:reply, {:ok, resp}, update_state(state, user_ctx)}
-        else
-          error ->
-            Logger.error("Error on parse response #{inspect(error)}")
-            {:reply, {:error, :invalid_content}, state, :hibernate}
-        end
-
-      {:error, timeout} ->
-        Logger.error("User Function Actor Invocation Timeout Error")
-        {:reply, {:error, timeout}, state, :hibernate}
-
-      {:error, reason} ->
-        Logger.error("User Function Actor Invocation Unknown Error: #{inspect(reason)}")
-        {:reply, {:error, reason}, state, :hibernate}
-
-      error ->
-        Logger.error("User Function Actor Invocation Unknown Error")
-        {:reply, {:error, error}, state, :hibernate}
-    end
+    ActorInvocation.new(
+      actor_name: name,
+      actor_system: actor_system,
+      command_name: command,
+      value: payload,
+      current_context: Context.new()
```

Comment:
You can remove the whole function pattern matching here and just in one function pass `Context.new(state: current_state)` it should be the same and you could avoid matching